### PR TITLE
OTel Scope.close() warning improvement

### PR DESCRIPTION
**Pull request by brunobat**
Helps to debug this: https://github.com/quarkusio/quarkus/issues/25102

### CODE REVIEW

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java [hunk 1]
```diff
@@ -54,22 +59,41 @@ public static Map<String, String> convertKeyValueListToMap(List<String> headers)
...
     public static void setMDCData(Context context, io.vertx.core.Context vertxContext) {
+        setMDCData(getSpanData(context), vertxContext);
+    }
+
+    public static void setMDCData(Map<String, String> spanData, io.vertx.core.Context vertxContext) {
+        if (spanData != null && !spanData.isEmpty()) {
+            spanData.entrySet().stream()
+                    .filter(entry -> SPAN_DATA_KEYS.contains(entry.getKey())) // filter out unknown keys
+                    .forEach(entry -> VertxMDC.INSTANCE.put(entry.getKey(), entry.getValue(), vertxContext));
+        }
```
Comment (geoand) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java:70]: We generally try pretty hard to not have lambdas in runtime code

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java [hunk 2]
```diff
@@ -16,6 +20,7 @@ public final class OpenTelemetryUtil {
     public static final String SPAN_ID = "spanId";
     public static final String SAMPLED = "sampled";
     public static final String PARENT_ID = "parentId";
+    private static final Set<String> SPAN_DATA_KEYS = new HashSet<>(Arrays.asList(TRACE_ID, SPAN_ID, SAMPLED, PARENT_ID));
```
Comment (famod) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java:23]:
```suggestion
    private static final Set<String> SPAN_DATA_KEYS = Set.of(TRACE_ID, SPAN_ID, SAMPLED, PARENT_ID);
```
:slightly_smiling_face:

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java (hunk 1 above)
Comment (brunobat) [reply extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java:70]: Sorry...

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java (hunk 2 above)
Comment (brunobat) [reply extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java:23]: ok

**Review by famod — APPROVED** [review]
LGTM!

Do we need @radcortez' blessing? 🙂

**Review by geoand — APPROVED** [review]

**Review by radcortez — APPROVED** [review]

**Comment by famod** [comment]
Thanks everyone!

**Pull request merged**